lib/miq_automation_engine/service_models/miq_ae_service_generic_object.rb

```diff
@@ -25,10 +33,23 @@ def ae_user_identity
       ar_method { @object.ae_user_identity(@ae_user, @ae_user.current_group, @ae_user.current_tenant) }
     end
 
+    def define_ae_method_from_generic_object(method_name)
+      self.instance_eval do
+        define_singleton_method(method_name) do |*args|
+          wrap_results(@object.send(:_call_automate, method_name, *args))
+        end
+      end
+    end
+
     def method_missing(method_name, *args)
       ae_user_identity unless @ae_user
       args = convert_params_to_ar_model(args)
-      results = object_send(method_name, *args)
+      if @object.property_method_defined?(method_name)
```

Comment:
We did not want to keep the `method_missing` or `respond_to_missing?` methods in this file at all.  All Generic Object attributes and methods should be created in the `initialize` method which removes the need for this logic.

Also, the `ae_user_identity` and `convert_params_to_ar_model` logic would move to the dynamic method definition.